prow/github/types.go

```diff
@@ -943,10 +943,13 @@ type DraftReviewComment struct {
 	Body     string `json:"body"`
 }
 
-// Content is some base64 encoded github file content
+// Content contains information about a github file/directory.
 type Content struct {
 	Content string `json:"content"`
 	SHA     string `json:"sha"`
+	Type    string `json:"type"`
+	Name    string `json:"name"`
+	Path    string `json:"path"`
```

Comment:
I wonder if we should have different struct types for different responses. Looking at the different responses in https://docs.github.com/en/free-pro-team@latest/rest/reference/repos#get-repository-content--code-samples it seem that e.g. for directories the `Content` member would not exist, so code that wants to work with `Content`  struct would always need to check the `Type` member which is IMO not that great.

So maybe we should have a `DirectoryContent` struct type that `GetDirectory` would always return?

Reply:
Sure, that would be cleaner. I have added `DirectoryContent` struct type now.